Docs: Manage multi-team access (#111824)
grafana/grafana: docs/sources/setup-grafana/configure-security/manage-single-access.md

---
aliases:
  - ../../enterprise/manage-single-access/
description: Manage multi-team access in a single Grafana instance
keywords:
  - grafana
  - rbac
  - lbac
  - auth
  - access
  - teams
labels:
  products:
    - cloud
    - enterprise
    - oss
title: Manage multi-team access in a single Grafana instance
weight: 1200

> ---

Author: @jtvdez @irenerl24 - can you guys check I added the refs correcty? TIA!

# Manage multi-team access in a single Grafana instance

If your organization has multiple teams using Grafana, you can use a single Grafana deployment (OSS/Enterprise) or a single Grafana Cloud stack to manage access across teams using roles and folders. This approach reduces complexity, simplifies identity and access management, and facilitates cross-team collaboration.

## Benefits

By using a single Grafana instance to manage access, you can:

- Implement a unified SSO, establishing clear permissions.
- Reduce setup and maintenance work, avoiding multi-stack complexity.

> - Centralize configuration and plugin management.

Reviewer: ```suggestion
- Centralize plugin configuration and management.
```

- Ensure teams can access the right dashboards and data, avoiding stepping on or overwriting each other’s work.
- Enable collaboration across teams. Teams are not isolated in silos and can discover and collaborate with each other’s work.
- Optimize resource management. With shared spaces, like an “Everyone” folder, you can publish executive dashboards or cross-team metrics that all groups can benefit from, without duplicating it across stacks.

## Example: Three teams, one stack

Let’s imagine a setup with three teams:

- Team A builds product features and needs autonomy with their own dashboards and data sources.
- Team B handles data engineering and needs autonomy with their own dashboards.
- Team C is the observability team and the admins of the Grafana stack.

> Follow these suggested steps to structure, configure and set permissions to access data in your Grafana instance:

Reviewer: ```suggestion
Follow these suggested steps to structure, configure, and set permissions to access data in your Grafana instance:
```

> 1. [Before you begin](#before-you-begin)

Reviewer: Could this be an H3? Basically:
### Before you begin

{{< /admonition >}}

### Create teams and configure user access

After you’ve deployed your Grafana instance:

- To follow the example in this doc, create three [Grafana Teams](../../../administration/team-management/configure-grafana-teams/#create-a-grafana-team) and add them to the Grafana instance.
- Determine the [RBAC](ref:rbac) strategy for your organization. RBAC extends default Grafana roles, provides more granular access rights, and simplifies how to grant, modify, or revoke user access to Grafana resources, such as users and reports.
- Assign each user to the [relevant team](../../../administration/user-management/manage-org-users/). By default [new users](../../configure-grafana/#auto_assign_org) are granted the **Viewer** role.
- Assign the [**Admin** role](ref:rbac-assign) to Team C so that they can manage all resources in the instance.

### Design a folder structure to match your access needs

To design a [folder](ref:create-folder) setup that helps users quickly understand where to go, what they can access, and what they can manage:

> - Create an “Everyone” folder for shared items that all teams can manage.

Reviewer 2: ```suggestion
- Create an “Everyone” folder for shared items that all teams can manage (ie. grant them admin access to that folder).
```

{{< figure src="/media/docs/grafana/oac/AccessTeams02.png" max-width="750px" alt="Teams and data sources in the stack, and the related permissions Team A and Team B have been granted" >}}

#### Resources at an instance level

Some Grafana resources, such as service accounts, alert contact points, [Fleet Management collectors](https://grafana.com/docs/grafana-cloud/send-data/fleet-management/), and other feature resources, are not linked to teams but are managed at the stack level. For these type of resources, assign fixed roles to teams carefully.

For example, users working in [Frontend Observability](https://grafana.com/docs/grafana-cloud/monitor-applications/frontend-observability/) need a writer fixed role so that they can create and manage services.

{{< figure src="/media/docs/grafana/oac/AccessTeams03.png" max-width="750px" alt="Grafana Cloud Frontend Observability resources in the stack, and the related permissions Team A have been granted" >}}

### Scale access management with Terraform and SSO

After you've made sure the model is working, you can codify it.

> You can add any new users to your Grafana instance with an Identity Provider through [SCIM](../../configure-security/configure-scim-provisioning/). Use [role sync](../../configure-security/configure-authentication/saml/configure-saml-team-role-mapping/#configure-role-sync-for-saml) to ensure everyone is automatically assigned to the right team based on their membership, including those with the Admin role.

Reviewer 2: ```suggestion
You can add any new users to your Grafana instance with an Identity Provider through [SCIM](../../configure-security/configure-scim-provisioning/). Use [role sync](../../configure-security/configure-authentication/saml/configure-saml-team-role-mapping/#configure-role-sync-for-saml) to automatically assign users the correct basic role (Viewer, Editor, or Admin) based on their mapped attributes in the IdP..
```
